src/trove/string-dict.scrbl

```diff
@@ -423,4 +530,102 @@ check:
 end
 }
 
+@msd-method["merge-now"
+  #:contract (a-arrow (MSD-of "a") (MSD-of "a") No)
+  #:args (list (list "self" #f) (list "other" #f))
+  #:return No
+]
+
+Modifies the mutable string-dict to include the keys from the @pyret{other}
+mutable-string-dict. Choose a key's value to be the one it has in the @pyret{other}
+string-dict, if it has one; otherwise it keeps the value it originally had.
+Returns nothing.
+
+@examples{
+check:
+  msd1 = [mutable-string-dict: "a", 5, "c", 4]
+  msd2 = [mutable-string-dict: "a", 10, "b", 6]
+  msd1.get-value-now("a") is 5
+  msd1.get-value-now("b") raises "Key b not found"
+  msd1.get-value-now("c") is 4
+  msd1.merge-now(msd2) is nothing
+  msd1.get-value-now("a") is 10
+  msd1.get-value-now("b") is 6
+  msd1.get-value-now("c") is 4
+end
+}
+
+@msd-method["keys-list-now"
+  #:contract (a-arrow (MSD-of "a") (L-of S))
+  #:args (list (list "self" #f))
+  #:return (L-of S)
+]
+
+Returns the list of keys in the mutable string-dict, in some order.
+
+@examples{
+check:
+  msd2 = [mutable-string-dict: "a", 10, "b", 6]
+  msd2.keys-list-now() is [list: "a", "b"]
+end
+}
+
+@msd-method["map-keys-now"
+  #:contract (a-arrow (MSD-of "a") (a-arrow "a" "b") (L-of "b"))
+  #:args (list (list "self" #f) (list "f" #f))
+  #:return (L-of "b")
+]
+
+Modify the mutable string-dict so that each key is now
+associated with the result of applying
+the function @pyret{f} to its original value.
+
+@examples{
+fun one-of(ans, elts):
+  is-some(for find(elt from elts):
+    ans == elt
+  end)
+end
+
+check:
+  msd2 = [mutable-string-dict: "a", 10, "b", 6]
+  msd2.map-keys-now(lam(x): string-append(x, x) end) is%(one-of)
+    [list: [list: "aa", "bb"], [list: "bb", "aa"]]
+end
+}
+
+@msd-method["fold-keys-now"
+  #:contract (a-arrow (MSD-of "a") (a-arrow S "b") "b")
+  #:args (list (list "self" #f) (list "f" #f) (list "init" #f))
+  #:return "b"
+]
+
+Returns the result of folding the function @pyret{f} across the keys in the
+mutable string-dict, in some order, starting with @pyret{init} as the base value.
+
+@examples{
+check:
+  msd2 = [mutable-string-dict: "a", 10, "b", 6]
+  msd2.fold-keys-now(lam(x, acc): acc + string-length(x) end, 0) is 2
+end
+}
+
+@msd-method["each-key-now"
+  #:contract (a-arrow (MSD-of "a") (a-arrow S No) No)
+  #:args (list (list "self" #f) (list "f" #f))
+  #:return No
+]
+
+Calls the function @pyret{f} on each key in the mutable string-dict,
```

Comment:
ditto

Reply:
Thanks -- made your suggested changes. When having the description of `each-key` refer to `map-keys`, I tried using `@pyret-id{map-keys}` rather than `@pyret{map-keys}` to make it a hyperlink, but it doesn't seem to work. Is there a Pyret scribble quickref that will show how to ensure that `@pyret-id{...}` does link.

Reply:
Use `@pyret-method["StringDict"]{each-key}` to link to the `each-key` method of the `StringDict` data type.